### PR TITLE
Enum Value Validator

### LINKED ISSUE #642: Create an enum value validator
As an AaC Developer, I want an enum value validator, so that I know when an enum value is used in a field it is defined by an enum type. 

## AC:

- [ ] A new validator is created for enum values
- [ ] A unit test for the validator is created
- [ ] The validator is added to the pipeline core validators if necessary
- [ ] The validator documentation is updated to cover the new validator

### PR DESCRIPTION
**Pull request by Coffee2Bits**
# Description
Added the Enum Value Validator that checks whether the enum value in a field is one of the enum values defined as a valid value.

# Linked Items:
Please include the items resolved by and/or associated with this Pull Request using the following syntax to link them for closure upon completion.
Closes #642 

## Type of change
- [X] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Feature improvement

### Feature Additions
Include pertinent changes as a result of adding the new feature.
- Added new validation plugin

### Bug Fixes
Include pertinent changes of fixing the bug.
- None

### Breaking Changes
Include pertinent changes that qualify as breaking changes. Be sure to include information about the impact to end users and additional information for addressing the break if more than a version transition is necessary for mitigating.
- None

### Feature Improvement Changes
Include pertinent changes that are improvements to existing features.
- None

# Checklist:
- [x] Updated documentation accordingly.
- [x] My changes generate no new warnings.
- [x] Have updated new and existing unit tests accordingly.
- [x] Linked associated items to be closed.
- [x] Have you bumped the version if there will be a release?

### CODE REVIEW

File: python/src/aac/validate/_validate.py [hunk 1]
```diff
@@ -216,43 +215,79 @@ def _validate_fields(
...
+
+    if not enum_type_definition:
+        logging.error(f"Unable to find enum definition for type {field_type}. Can't validate field defined as '{field_type}' in '{definition.name}'")
+    else:
+        validations = enum_type_definition.get_validations() or []
+        validation_plugins = language_context.get_definition_validations()
+
+        for validation in validations:
+            validation_name = validation.get(DEFINITION_FIELD_NAME)
+            # This line will throw an exception if there is a mismatch between the plugin name and validation name. We need to better handle this.
```
Comment (Coffee2Bits) [inline python/src/aac/validate/_validate.py:271]:
```suggestion
```

**Review by cvchaparro — CHANGES_REQUESTED** [review]
Just a few minor suggestions.

File: python/src/aac/lang/language_context.py [hunk 2]
```diff
@@ -298,11 +298,24 @@ def get_primitive_types(self) -> list[str]:
             A list of strings, one entry for each primitive type defined in the LanguageContext.
 
             These types may differ from those provided by the core spec since the LanguageContext applies definitions
-            from active plugins and user files, which may extend the set of root keys.
+            from active plugins and user files.
             See :py:func:`aac.spec.get_primitives()` for the list of root keys provided by the unaltered core AaC DSL.
         """
         return deepcopy(self.get_primitives_definition().get_values()) or []
 
+    def get_enum_types(self) -> list[str]:
```
[the remark this answers is not in the document]
Comment (Coffee2Bits) [reply python/src/aac/lang/language_context.py:306]: Done

**Review by cvchaparro — APPROVED** [review]
LGTM

File: python/tests/plugins/validators/test__validate_enum_values.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+        test_active_context = get_active_context(reload_context=True)
+        behavior_type_definition = test_active_context.get_definition_by_name(DEFINITION_NAME_BEHAVIOR_TYPE)
+
+        actual_result = validate_enums(test_model, behavior_type_definition, test_active_context)
+
+        self.assertEqual(expected_result.is_valid(), actual_result.is_valid())
+
+    def test_validate_enum_values_invalid(self):
+        # Create a model with a behavior to test that the validation is valid when a valid BehaviorType enum value is used.
```
Comment (lizzcondrey) [inline python/tests/plugins/validators/test__validate_enum_values.py:27]:
```suggestion
        # Create a model with a behavior to test that the validation is invalid when an invalid BehaviorType enum value is used.
```

File: python/tests/plugins/validators/test__validate_enum_values.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+        test_active_context = get_active_context(reload_context=True)
+        behavior_type_definition = test_active_context.get_definition_by_name(DEFINITION_NAME_BEHAVIOR_TYPE)
+
+        actual_result = validate_enums(test_model, behavior_type_definition, test_active_context)
+
+        self.assertFalse(actual_result.is_valid())
+
+    def test_validated_definition_catches_invalid(self):
+        # Create a model with a behavior to test that the validation is valid when a valid BehaviorType enum value is used.
```
Comment (lizzcondrey) [inline python/tests/plugins/validators/test__validate_enum_values.py:39]:
```suggestion
        # Create a model with a behavior to test that the validation is caught in definition validation when an invalid BehaviorType enum value is used.
```

**Review by lizzcondrey — APPROVED** [review]

**Pull request merged**